app/controllers/users/invitations_controller.rb

```diff
@@ -26,6 +26,7 @@ def validate_invited_user
   end
 
   def user_is_invalid?
+    delete_user_record if invited_user_already_exists? && invited_user_not_confirmed?
```

Comment:
This looks like correct logic but in a weird part of this file. The method name `def user_is_invalid?` doesn't relate to checking if this user is an unconfirmed user.

I would suggest pulling it out so its more explicit, kind of how `before_action :delete_user_record, if: :resending_invite?, only: :create` makes a lot of sense.

One idea could be:
```
before_action :delete_user_record, if: :user_is_unconfirmed_and_without_org?, only: :create

def user_is_unconfirmed_and_without_org?
  invited_user_already_exists? && invited_user_not_confirmed?
end
```

This is just an example. The point is its currently quite hard to understand whats going on and `def user_is_invalid?` is the wrong place for the logic

Reply:
`invited_user_already_exists? && invited_user_not_confirmed?` doesn't cover the case of a user that is unconfirmed but has an organisation.